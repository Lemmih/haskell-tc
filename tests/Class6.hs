module Class6 where

class Plus a where
  plus :: a -> a -> a

data Maybe a = Nothing | Just a

instance Plus (Maybe a) where
  plus = \a b -> Nothing
