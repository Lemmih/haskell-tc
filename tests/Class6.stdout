Bindings:
:: âˆ€ a. Class6.Plus a â‡’ a â†’  a â†’  a
    [4mplus[0m :: a -> a -> a
:: * â†’  *
  data [4mMaybe[0m a = Nothing | Just a
:: Î›a. Class6.Maybe a
  data Maybe a = [4mNothing[0m | Just a
:: Î›a. a â†’  Class6.Maybe a
  data Maybe a = Nothing | [4mJust[0m a
:: Class6.Maybe sk_0_a
    plus = \[4ma[0m b -> Nothing
:: Class6.Maybe sk_0_a
    plus = \a [4mb[0m -> Nothing

Proofs:
coercion: âˆ€ a. Class6.Maybe a
  instance [4mPlus (Maybe a)[0m where
coercion: âˆ€ sk_0_a. Class6.Maybe sk_0_a â†’  Class6.Maybe sk_0_a â†’  Class6.Maybe sk_0_a
    [4mplus = \a b -> Nothing[0m
coercion: (âˆ€ a. Class6.Maybe a) @ sk_0_a
    plus = \a b -> [4mNothing[0m
